pkg/noderesourcetopology/cache/indexer.go

```diff
@@ -0,0 +1,236 @@
+	defer nni.rwLock.RUnlock()
+	data, ok := nni.nodeToPodsMap[nodeName]
+	if !ok {
+		klog.V(6).InfoS("nrtcache: nni GET ", "logID", logID, "node", nodeName, "pods", "NONE")
+		return []types.NamespacedName{}, nil
+	}
+
+	objs := make([]types.NamespacedName, 0, len(data))
+	for _, item := range data {
+		objs = append(objs, item.namespacedName)
+	}
+	klog.V(6).InfoS("nrtcache: nni GET ", "logID", logID, "node", nodeName, "pods", namespacedNameListToString(objs))
+	return objs, nil
+}
+
+func (nni *nodeNameIndexer) TrackReservedPod(pod *corev1.Pod, nodeName string) {
+	nni.rwLock.Lock()
+	defer nni.rwLock.Unlock()
+	nni.podUIDToCandidateNodeMap[pod.UID] = nodeName
+	nni.insertPod(pod, nodeName, podObservedReserved)
+	klog.V(6).InfoS("nrtcache: nni RES ", "logID", klog.KObj(pod), "node", nodeName, "podUID", pod.UID)
+}
+
+func (nni *nodeNameIndexer) UntrackReservedPod(pod *corev1.Pod, nodeName string) {
+	nni.rwLock.Lock()
+	defer nni.rwLock.Unlock()
+	delete(nni.podUIDToCandidateNodeMap, pod.UID)
+	obsState, ok := nni.getPodObservedState(pod, nodeName)
+	if ok && obsState == podObservedBound {
+		delete(nni.nodeToPodsMap[nodeName], pod.UID)
+	}
+	klog.V(6).InfoS("nrtcache: nni UNR ", "logID", klog.KObj(pod), "node", nodeName, "podUID", pod.UID, "obsState", obsState)
+}
+
+func (nni *nodeNameIndexer) addPod(pod *corev1.Pod) {
+	if pod.Spec.NodeName == "" {
+		klog.V(4).InfoS("nrtcache: nni WARN", "logID", klog.KObj(pod), "node", "EMPTY", "podUID", pod.UID, "phase", pod.Status.Phase)
+		return
+	}
+	if pod.Status.Phase != corev1.PodRunning {
+		// We should only consider pod which are consuming resources. But it's also paramount to take into account what the
+		// node side can do. The primary source of data for node agents is the podresources API, and the podresources API
+		// cannot do any filtering based on pod runtime status. Accessing this data from other sources (e.g. apiserver, runtime)
+		// is either hard (runtime) or should be avoided for performance reasons (apiserver). So the only compromise is
+		// to overshoot and - incorrectly - account for ALL the pods reported on the node. Let's at least acknowledge
+		// this fact and be pretty loud when it occurs.
+		klog.V(4).InfoS("nrtcache: nni WARN", "logID", klog.KObj(pod), "node", pod.Spec.NodeName, "podUID", pod.UID, "phase", pod.Status.Phase)
+		// intentional fallback
+	}
+	nni.rwLock.Lock()
+	defer nni.rwLock.Unlock()
+	klog.V(6).InfoS("nrtcache: nni ADD ", "logID", klog.KObj(pod), "node", pod.Spec.NodeName, "podUID", pod.UID)
+	nni.insertPod(pod, pod.Spec.NodeName, podObservedBound)
+}
+
+func (nni *nodeNameIndexer) deletePod(pod *corev1.Pod) {
+	nni.rwLock.Lock()
+	defer nni.rwLock.Unlock()
+
+	nodeName := pod.Spec.NodeName
+	wasReserved := false
+	if nodeName == "" {
+		// we're deleting a pod which passed the reserve stage but never got to the binding stage.
+		// Should be unlikely, but can happen!
+		nodeName = nni.podUIDToCandidateNodeMap[pod.UID]
+		wasReserved = true
```

Comment:
nit:

```suggestion
		nodeName, wasReserved = nni.podUIDToCandidateNodeMap[pod.UID]
```

Reply:
right, and actually more correct. Fixed!